### ISSUE TITLE
WS encoding

### ISSUE THREAD

**Issue by computcon** [haxball/haxball-issues#1771]
How does the arraybuffer encoding of the ```p2p.haxball.com``` server and ```haxball.com/rs/api/list``` work and why is it used instead of regular text?

**Comment by wxyz-abcd** [comment]
We use ArrayBuffer's to store, send and/or receive data more efficiently. The data can be accessed and/or changed efficiently using DataView's. There are around 10 different standard data type encodings in the DataView class. 
Let's say you want to send 2 floating point values with different length: [5.3917218335251, -27.123531126251]. As you can see, you need a length-15 buffer and a length-16 buffer to write these two numbers. But if we use the ArrayBuffer encoding, the necessary length will be 4 for each number.

**Comment by Skarmunds** [comment]
Yep, put example and bytes overview in readme. https://github.com/Skarmunds/haxball_rooms_api

**Comment by wxyz-abcd** [comment]
> Yep, put example and bytes overview in readme. https://github.com/Skarmunds/haxball_rooms_api

Nicely done. Maybe I should do the same for all haxball networking messages, since i've decoded almost all of it. :)))

**Comment by i13k** [comment]
> > Yep, put example and bytes overview in readme. https://github.com/Skarmunds/haxball_rooms_api
> 
> Nicely done. Maybe I should do the same for all haxball networking messages, since i've decoded almost all of it. :)))

I'd love to see that!
